src/time.c

```diff
@@ -685,11 +685,19 @@ get_thread_usage_ts(struct timespec *ts)
 
 #else /* __MPLS_TARGET_OSVER >= 101000 */
 
-/* Get the CPU usage of the current thread via syscall, in nanoseconds. */
+/*
+ * Get the CPU usage of the current thread via syscall, in nanoseconds.
+ *
+ * We do a trick here to override the return type of syscall to avoid overflows.
+ */
 static inline uint64_t
 get_thread_usage_ns(void)
 {
-  uint64_t mach_time = syscall(SYS_thread_selfusage);
+  static uint64_t (*syscall64)(int, ...) = NULL;
+  if (!syscall64)
+    syscall64 = dlfcn(RTLD_NEXT, "syscall");
```

Comment:
Should this be `dlsym`? If not, where is `dlfcn` defined? If so, `dlfcn.h` probably needs to be included.

Reply:
Yes, the function is called `dlsym`, which is declared in the `dlfcn.h` header, though I'm not yet convinced that it's necessary to use it at all in this context, and even if it were, there are macros here for that.

As I said, I'll look into it.